### ISSUE TITLE
Spyder 1.1.0rc2 only starts as root in Ubuntu Lucid (10.04)

### ISSUE THREAD

**Issue by spyder-issues-migrator** [spyder-ide/spyder#240]
_From [jfalo...@gmail.com](https://code.google.com/u/102909429257844123103/) on 2010-06-23T01:31:09Z_

Runs flawless as root, but I cannot start it as regular user. I had the same problem with 1.1.0rc1. Here's the traceback:

Traceback (most recent call last):
  File "/usr/local/bin/spyder.py", line 4, in \<module>
    import pkg_resources
  File "/usr/lib/python2.6/dist-packages/pkg_resources.py", line 2655, in \<module>
    working_set.require(__requires__)
  File "/usr/lib/python2.6/dist-packages/pkg_resources.py", line 648, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/usr/lib/python2.6/dist-packages/pkg_resources.py", line 546, in resolve
    raise DistributionNotFound(req)
pkg_resources.DistributionNotFound: spyder==1.1.0rc2

_Original issue: http://code.google.com/p/spyderlib/issues/detail?id=240_

**Comment by spyder-issues-migrator** [comment]
_From [jfalo...@gmail.com](https://code.google.com/u/102909429257844123103/) on 2010-06-28T23:49:41Z_

After a recent update to spyder 1.1.1, the error still continues.

Traceback (most recent call last):
  File "/usr/local/bin/spyder.py", line 4, in \<module>
    import pkg_resources
  File "/usr/lib/python2.6/dist-packages/pkg_resources.py", line 2655, in \<module>
    working_set.require(__requires__)
  File "/usr/lib/python2.6/dist-packages/pkg_resources.py", line 648, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/usr/lib/python2.6/dist-packages/pkg_resources.py", line 546, in resolve
    raise DistributionNotFound(req)
pkg_resources.DistributionNotFound: spyder==1.1.1

**Comment by spyder-issues-migrator** [comment]
_From [pierre.raybaut](https://code.google.com/u/pierre.raybaut/) on 2010-06-29T00:54:00Z_

Apparently this is an issue related to setuptools (.egg files installer management library).

I suppose you have installed Spyder with easy_install?
Could you please post here the exact command line that you used to install Spyder?

**Comment by spyder-issues-migrator** [comment]
_From [jfalo...@gmail.com](https://code.google.com/u/102909429257844123103/) on 2010-06-29T00:59:04Z_

At first I downloaded the .egg file and installed myself using easy_install. I did this as root:

   easy_install spyder-1.1.0rc1-py2.6.egg

And later I've been updating spyder:

   easy_install -U spyder

All other software (python, etc.) has been installed from ubuntu repositories using aptitude.

**Comment by spyder-issues-migrator** [comment]
_From [pierre.raybaut](https://code.google.com/u/pierre.raybaut/) on 2010-06-29T01:06:43Z_

Could you try to remove completely Spyder and reinstall it directly with:
easy_install spyder
?

Thx!

**Comment by spyder-issues-migrator** [comment]
_From [jfalo...@gmail.com](https://code.google.com/u/102909429257844123103/) on 2010-06-29T01:19:38Z_

Ok, I have removed all the content in /usr/local/lib/python2.6/dist-packages and reinstalled spyder using easy_install:

Searching for spyder
Reading http://pypi.python.org/simple/spyder/ Reading http://spyderlib.googlecode.com Best match: spyder 1.1.1
Downloading http://spyderlib.googlecode.com/files/spyder-1.1.1.tar.gz Processing spyder-1.1.1.tar.gz
Running spyder-1.1.1/setup.py -q bdist_egg --dist-dir /tmp/easy_install-XlDF2a/spyder-1.1.1/egg-dist-tmp-4FuOoD
zip_safe flag not set; analyzing archive contents...
spyderlib.interpreter: module references __file__
spyderlib.config: module references __file__
spyderlib.widgets.editor: module references __file__
spyderlib.widgets.pylintgui: module references __file__
spyderlib.widgets.qteditor.qteditor: module references __file__
spyderlib.widgets.externalshell.pythonshell: module references __file__
spyderlib.widgets.externalshell.sitecustomize: module references __file__
spyderlib.widgets.externalshell.startup: module references __file__
spyderlib.widgets.externalshell.__init__: module references __file__
spyderlib.widgets.qscieditor.qscieditor: module references __file__
spyderlib.utils.iofuncs: module references __path__
spyderlib.utils.dochelpers: module MAY be using inspect.getsource
Adding spyder 1.1.1 to easy-install.pth file
Installing spyder script to /usr/local/bin

Installed /usr/local/lib/python2.6/dist-packages/spyder-1.1.1-py2.6.egg
Processing dependencies for spyder
Finished processing dependencies for spyder

But when I try to run spyder as regular user, I get the same error. Do I have to uninstall spyder in a different manner?

**Comment by spyder-issues-migrator** [comment]
_From [jfalo...@gmail.com](https://code.google.com/u/102909429257844123103/) on 2010-07-02T05:56:07Z_

Hi Pierre,

I just made a "clean install" in my laptop (running Ubuntu Lucid) and spyder 1.1.2 works fine.

I have removed spyder 1.1.1 from /usr/local/lib/python2.6/dist-packages and the scripts from  /usr/local/bin. I have also deleted my ~/.spyder folder and I have installed spyder 1.1.2 using easy install. 

Finally it is working now, I have noticed I had wrong permissions for the spyder script!

Sorry for the inconvenience and thank you for your help!!
